File: C#/Problems/Easy/Problem1313/Solution.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LeetCode.Problems.Easy.Problem1313
{
    public class Solution {
        public int[] DecompressRLElist(int[] nums) {
            var result = new List<int>();
            
            for (int i = 0; i < nums.Length; i += 2) {
                var freq = nums[i];
                var val = nums[i+1];
                
                while (freq > 0) {
                    result.Add(val);
                    freq--;
                }
            }
            
            return result.ToArray();
        }
    }
}

File: Golang/problems/easy/problem1313/solution.go
package solutions

// DecompressRLElist function is a solution for the
// Problem #1313. Decompress Run-Length Encoded List
// from leetcode.com
func DecompressRLElist(nums []int) []int {
	var result []int

	for i := 0; i < len(nums); i += 2 {
		var freq, val = nums[i], nums[i+1]
		for ; freq > 0; freq-- {
			result = append(result, val)
		}
	}
	return result
}
